Godot 3.2 Mono - Android build does not find shared Mono library

**Godot version:**
Godot 3.2 RC1 Mono Custom build from Godot repo branch 3.2

**OS/device including version:**
Mac

**Issue description:**
I am using Mac. I decided to do custom build to reduce Apk size.

I ran " scons p=osx tools=yes module_mono_enabled=yes mono_glue=yes -j 6". And it worked successfully. I created Godot mono binary for Mac from Godot repo branch 3.2.

I crossed compile the Mono runtime for the target architectures by using the page: https://github.com/godotengine/godot-mono-builds. I can see all 4 mono releases for armv7, arm64v8, x86 and x86_64.

<img width="618" alt="Screen Shot 2020-02-24 at 13 38 56" src="https://user-images.githubusercontent.com/47486829/75156941-0a764000-570b-11ea-9a8a-371922fc0369.png">

Then i tried to run the command below:
`scons -j 6 platform=android --config=force target=release_debug android_arch=arm64v8 tools=no module_mono_enabled=yes deprecated=no minizip=no builtin_glew=no builtin_libmpcdec=no disable_advanced_gui=yes disable_3d=yes builtin_openssl=no module_openssl_enabled=no module_bmp_enabled=no builtin_libtheora=no builtin_libvorbis=no builtin_opus=no module_bullet_enabled=no module_csg_enabled=no builtin_speex=no builtin_squish=no builtin_zlib=no module_dds_enabled=no module_chibi_enabled=no module_enet_enabled=no module_gdnative_enabled=no module_gridmap_enabled=no module_hdr_enabled=no module_mbedtls_enabled=no module_jpg_enabled=no module_ik_enabled=no module_mobile_vr_enabled=no module_opus_enabled=no module_pvr_enabled=no module_recast_enabled=no module_regex_enabled=no module_squish_enabled=no module_mpc_enabled=no module_tga_enabled=no module_thekla_unwrap_enabled=no module_theora_enabled=no module_tinyexr_enabled=no module_vorbis_enabled=no module_webm_enabled=no module_websocket_enabled=no module_pbm_enabled=no module_speex_enabled=no module_etc_enabled=no mono_prefix=$home/mono-installs/android-arm64-v8a-release
`

But it failed even if i set mono_prefix
`scons: Reading SConscript files ...
Building for Android (arm64v8)
WARNING: android_arch=arm64v8 is not supported by ndk_platform lower than android-21; setting ndk_platform=android-21
Checking for `thread_local` support... supported
Found Mono root directory: /mono-installs/android-arm64-v8a-release
RuntimeError: Could not find mono library in: /mono-installs/android-arm64-v8a-release/lib:
  File "/Users/user/Documents/Development/Mobile/Godot/git/godot/SConstruct", line 521:
    SConscript("modules/SCsub")
  File "/usr/local/lib/scons/SCons/Script/SConscript.py", line 660:
    return method(*args, **kw)
  File "/usr/local/lib/scons/SCons/Script/SConscript.py", line 597:
    return _SConscript(self.fs, *files, **subst_kw)
  File "/usr/local/lib/scons/SCons/Script/SConscript.py", line 286:
    exec(compile(scriptdata, scriptname, 'exec'), call_stack[-1].globals)
  File "/Users/user/Documents/Development/Mobile/Godot/git/godot/modules/SCsub", line 17:
    SConscript(x + "/SCsub")
  File "/usr/local/lib/scons/SCons/Script/SConscript.py", line 660:
    return method(*args, **kw)
  File "/usr/local/lib/scons/SCons/Script/SConscript.py", line 597:
    return _SConscript(self.fs, *files, **subst_kw)
  File "/usr/local/lib/scons/SCons/Script/SConscript.py", line 286:
    exec(compile(scriptdata, scriptname, 'exec'), call_stack[-1].globals)
  File "/Users/user/Documents/Development/Mobile/Godot/git/godot/modules/mono/SCsub", line 35:
    mono_configure.configure(env, env_mono)
  File "/Users/user/Documents/Development/Mobile/Godot/git/godot/modules/mono/build_scripts/mono_configure.py", line 204:
    raise RuntimeError('Could not find mono library in: ' + mono_lib_path)`


android-arm64-v8a-release folder structure:
<img width="594" alt="Screen Shot 2020-02-24 at 13 09 39" src="https://user-images.githubusercontent.com/47486829/75157036-3e516580-570b-11ea-89da-ed76e91165ba.png">

lib folder in android-arm64-v8a-release folder:
<img width="600" alt="Screen Shot 2020-02-24 at 13 41 17" src="https://user-images.githubusercontent.com/47486829/75157133-6ccf4080-570b-11ea-884e-2c039aa167a8.png">

Is there any issue with compiling Android export template for Godot Mono 3.2?

I have another issue at 100%:

`[100%] progress_finish(["progress_finish"], [])
[100%] Linking Static Library ==> modules/freetype/libfreetype_builtin.android.opt.debug.armv7.neon.a
[100%] Ranlib Library         ==> modules/freetype/libfreetype_builtin.android.opt.debug.armv7.neon.a
[100%] Linking Static Library ==> core/libcore.android.opt.debug.armv7.neon.a
Ranlib Library         ==> core/libcore.android.opt.debug.armv7.neon.a
[100%] Linking Shared Library ==> bin/libgodot.android.opt.debug.armv7.neon.so
[100%] clang++: warning: argument unused during compilation: '-dynamiclib' [-Wunused-command-line-argument]
/Users/user/Library/Android/sdk/ndk-bundle/toolchains/arm-linux-androideabi-4.9/prebuilt/darwin-x86_64/lib/gcc/arm-linux-androideabi/4.9.x/../../../../arm-linux-androideabi/bin/ld: error: cannot find -liconv
/Users/user/Library/Android/sdk/ndk-bundle/toolchains/arm-linux-androideabi-4.9/prebuilt/darwin-x86_64/lib/gcc/arm-linux-androideabi/4.9.x/../../../../arm-linux-androideabi/bin/ld: error: cannot find -lpthread
clang++: error: linker command failed with exit code 1 (use -v to see invocation)
scons: *** [bin/libgodot.android.opt.debug.armv7.neon.so] Error 1
scons: building terminated because of errors.`

I do have NDK-R21.

I ran the command below:

`/Users/user/Library/Android/sdk/ndk-bundle/toolchains/aarch64-linux-android-4.9/prebuilt/darwin-x86_64/lib/gcc/aarch64-linux-android/4.9.x/../../../../aarch64-linux-android/bin/ld -lpthread -verbose
`
`/Users/user/Library/Android/sdk/ndk-bundle/toolchains/aarch64-linux-android-4.9/prebuilt/darwin-x86_64/lib/gcc/aarch64-linux-android/4.9.x/../../../../aarch64-linux-android/bin/ld -liconv -verbose`

It seem that Mac does not have libpthread and libiconv

Full Output for libpthread:

GNU ld (GNU Binutils) 2.27.0.20170315
  Supported emulations:
   aarch64linux
   aarch64elf
   aarch64elf32
   aarch64elf32b
   aarch64elfb
   armelf
   armelfb
   aarch64linuxb
   aarch64linux32
   aarch64linux32b
   armelfb_linux_eabi
   armelf_linux_eabi
using internal linker script:
==================================================
/* Script for -z combreloc: combine and sort reloc sections */
/* Copyright (C) 2014-2016 Free Software Foundation, Inc.
   Copying and distribution of this script, with or without modification,
   are permitted in any medium without royalty provided the copyright
   notice and this notice are preserved.  */
OUTPUT_FORMAT("elf64-littleaarch64", "elf64-bigaarch64",
	      "elf64-littleaarch64")
OUTPUT_ARCH(aarch64)
ENTRY(_start)
SEARCH_DIR("=/Volumes/Android/buildbot/src/android/binutils/out/binutils/darwin/arm64/install/binutils-arm64-darwin/aarch64-linux-android/lib64"); SEARCH_DIR("=/usr/local/lib64"); SEARCH_DIR("=/lib64"); SEARCH_DIR("=/usr/lib64"); SEARCH_DIR("=/Volumes/Android/buildbot/src/android/binutils/out/binutils/darwin/arm64/install/binutils-arm64-darwin/aarch64-linux-android/lib"); SEARCH_DIR("=/usr/local/lib"); SEARCH_DIR("=/lib"); SEARCH_DIR("=/usr/lib");
SECTIONS
{
  /* Read-only sections, merged into text segment: */
  PROVIDE (__executable_start = SEGMENT_START("text-segment", 0x400000)); . = SEGMENT_START("text-segment", 0x400000) + SIZEOF_HEADERS;
  .interp         : { *(.interp) }
  .note.gnu.build-id : { *(.note.gnu.build-id) }
  .hash           : { *(.hash) }
  .gnu.hash       : { *(.gnu.hash) }
  .dynsym         : { *(.dynsym) }
  .dynstr         : { *(.dynstr) }
  .gnu.version    : { *(.gnu.version) }
  .gnu.version_d  : { *(.gnu.version_d) }
  .gnu.version_r  : { *(.gnu.version_r) }
  .rela.dyn       :
    {
      *(.rela.init)
      *(.rela.text .rela.text.* .rela.gnu.linkonce.t.*)
      *(.rela.fini)
      *(.rela.rodata .rela.rodata.* .rela.gnu.linkonce.r.*)
      *(.rela.data .rela.data.* .rela.gnu.linkonce.d.*)
      *(.rela.tdata .rela.tdata.* .rela.gnu.linkonce.td.*)
      *(.rela.tbss .rela.tbss.* .rela.gnu.linkonce.tb.*)
      *(.rela.ctors)
      *(.rela.dtors)
      *(.rela.got)
      *(.rela.bss .rela.bss.* .rela.gnu.linkonce.b.*)
      *(.rela.ifunc)
    }
  .rela.plt       :
    {
      *(.rela.plt)
      PROVIDE_HIDDEN (__rela_iplt_start = .);
      *(.rela.iplt)
      PROVIDE_HIDDEN (__rela_iplt_end = .);
    }
  .init           :
  {
    KEEP (*(SORT_NONE(.init)))
  } =0
  .plt            : ALIGN(16) { *(.plt) *(.iplt) }
  .text           :
  {
    *(.text.unlikely .text.*_unlikely .text.unlikely.*)
    *(.text.exit .text.exit.*)
    *(.text.startup .text.startup.*)
    *(.text.hot .text.hot.*)
    *(.text .stub .text.* .gnu.linkonce.t.*)
    /* .gnu.warning sections are handled specially by elf32.em.  */
    *(.gnu.warning)
  } =0
  .fini           :
  {
    KEEP (*(SORT_NONE(.fini)))
  } =0
  PROVIDE (__etext = .);
  PROVIDE (_etext = .);
  PROVIDE (etext = .);
  .rodata         : { *(.rodata .rodata.* .gnu.linkonce.r.*) }
  .rodata1        : { *(.rodata1) }
  .eh_frame_hdr : { *(.eh_frame_hdr) *(.eh_frame_entry .eh_frame_entry.*) }
  .eh_frame       : ONLY_IF_RO { KEEP (*(.eh_frame)) *(.eh_frame.*) }
  .gcc_except_table   : ONLY_IF_RO { *(.gcc_except_table
  .gcc_except_table.*) }
  .gnu_extab   : ONLY_IF_RO { *(.gnu_extab*) }
  /* These sections are generated by the Sun/Oracle C++ compiler.  */
  .exception_ranges   : ONLY_IF_RO { *(.exception_ranges
  .exception_ranges*) }
  /* Adjust the address for the data segment.  We want to adjust up to
     the same address within the page on the next page up.  */
  . = DATA_SEGMENT_ALIGN (CONSTANT (MAXPAGESIZE), CONSTANT (COMMONPAGESIZE));
  /* Exception handling  */
  .eh_frame       : ONLY_IF_RW { KEEP (*(.eh_frame)) *(.eh_frame.*) }
  .gnu_extab      : ONLY_IF_RW { *(.gnu_extab) }
  .gcc_except_table   : ONLY_IF_RW { *(.gcc_except_table .gcc_except_table.*) }
  .exception_ranges   : ONLY_IF_RW { *(.exception_ranges .exception_ranges*) }
  /* Thread Local Storage sections  */
  .tdata	  : { *(.tdata .tdata.* .gnu.linkonce.td.*) }
  .tbss		  : { *(.tbss .tbss.* .gnu.linkonce.tb.*) *(.tcommon) }
  .preinit_array     :
  {
    PROVIDE_HIDDEN (__preinit_array_start = .);
    KEEP (*(.preinit_array))
    PROVIDE_HIDDEN (__preinit_array_end = .);
  }
  .init_array     :
  {
    PROVIDE_HIDDEN (__init_array_start = .);
    KEEP (*(SORT_BY_INIT_PRIORITY(.init_array.*) SORT_BY_INIT_PRIORITY(.ctors.*)))
    KEEP (*(.init_array EXCLUDE_FILE (*crtbegin.o *crtbegin?.o *crtend.o *crtend?.o ) .ctors))
    PROVIDE_HIDDEN (__init_array_end = .);
  }
  .fini_array     :
  {
    PROVIDE_HIDDEN (__fini_array_start = .);
    KEEP (*(SORT_BY_INIT_PRIORITY(.fini_array.*) SORT_BY_INIT_PRIORITY(.dtors.*)))
    KEEP (*(.fini_array EXCLUDE_FILE (*crtbegin.o *crtbegin?.o *crtend.o *crtend?.o ) .dtors))
    PROVIDE_HIDDEN (__fini_array_end = .);
  }
  .ctors          :
  {
    /* gcc uses crtbegin.o to find the start of
       the constructors, so we make sure it is
       first.  Because this is a wildcard, it
       doesn't matter if the user does not
       actually link against crtbegin.o; the
       linker won't look for a file to match a
       wildcard.  The wildcard also means that it
       doesn't matter which directory crtbegin.o
       is in.  */
    KEEP (*crtbegin.o(.ctors))
    KEEP (*crtbegin?.o(.ctors))
    /* We don't want to include the .ctor section from
       the crtend.o file until after the sorted ctors.
       The .ctor section from the crtend file contains the
       end of ctors marker and it must be last */
    KEEP (*(EXCLUDE_FILE (*crtend.o *crtend?.o ) .ctors))
    KEEP (*(SORT(.ctors.*)))
    KEEP (*(.ctors))
  }
  .dtors          :
  {
    KEEP (*crtbegin.o(.dtors))
    KEEP (*crtbegin?.o(.dtors))
    KEEP (*(EXCLUDE_FILE (*crtend.o *crtend?.o ) .dtors))
    KEEP (*(SORT(.dtors.*)))
    KEEP (*(.dtors))
  }
  .jcr            : { KEEP (*(.jcr)) }
  .data.rel.ro : { *(.data.rel.ro.local* .gnu.linkonce.d.rel.ro.local.*) *(.data.rel.ro .data.rel.ro.* .gnu.linkonce.d.rel.ro.*) }
  .dynamic        : { *(.dynamic) }
  .got            : { *(.got) *(.igot) }
  . = DATA_SEGMENT_RELRO_END (24, .);
  .got.plt        : { *(.got.plt)  *(.igot.plt) }
  .data           :
  {
    PROVIDE (__data_start = .);
    *(.data .data.* .gnu.linkonce.d.*)
    SORT(CONSTRUCTORS)
  }
  .data1          : { *(.data1) }
  _edata = .; PROVIDE (edata = .);
  __bss_start = .;
  __bss_start__ = .;
  .bss            :
  {
   *(.dynbss)
   *(.bss .bss.* .gnu.linkonce.b.*)
   *(COMMON)
   /* Align here to ensure that the .bss section occupies space up to
      _end.  Align after .bss to ensure correct alignment even if the
      .bss section disappears because there are no input sections.
      FIXME: Why do we need it? When there is no .bss section, we don't
      pad the .data section.  */
   . = ALIGN(. != 0 ? 64 / 8 : 1);
  }
  _bss_end__ = . ; __bss_end__ = . ;
  . = ALIGN(64 / 8);
  . = SEGMENT_START("ldata-segment", .);
  . = ALIGN(64 / 8);
  __end__ = . ;
  _end = .; PROVIDE (end = .);
  . = DATA_SEGMENT_END (.);
  /* Stabs debugging sections.  */
  .stab          0 : { *(.stab) }
  .stabstr       0 : { *(.stabstr) }
  .stab.excl     0 : { *(.stab.excl) }
  .stab.exclstr  0 : { *(.stab.exclstr) }
  .stab.index    0 : { *(.stab.index) }
  .stab.indexstr 0 : { *(.stab.indexstr) }
  .comment       0 : { *(.comment) }
  /* DWARF debug sections.
     Symbols in the DWARF debugging sections are relative to the beginning
     of the section so we begin them at 0.  */
  /* DWARF 1 */
  .debug          0 : { *(.debug) }
  .line           0 : { *(.line) }
  /* GNU DWARF 1 extensions */
  .debug_srcinfo  0 : { *(.debug_srcinfo) }
  .debug_sfnames  0 : { *(.debug_sfnames) }
  /* DWARF 1.1 and DWARF 2 */
  .debug_aranges  0 : { *(.debug_aranges) }
  .debug_pubnames 0 : { *(.debug_pubnames) }
  /* DWARF 2 */
  .debug_info     0 : { *(.debug_info .gnu.linkonce.wi.*) }
  .debug_abbrev   0 : { *(.debug_abbrev) }
  .debug_line     0 : { *(.debug_line .debug_line.* .debug_line_end ) }
  .debug_frame    0 : { *(.debug_frame) }
  .debug_str      0 : { *(.debug_str) }
  .debug_loc      0 : { *(.debug_loc) }
  .debug_macinfo  0 : { *(.debug_macinfo) }
  /* SGI/MIPS DWARF 2 extensions */
  .debug_weaknames 0 : { *(.debug_weaknames) }
  .debug_funcnames 0 : { *(.debug_funcnames) }
  .debug_typenames 0 : { *(.debug_typenames) }
  .debug_varnames  0 : { *(.debug_varnames) }
  /* DWARF 3 */
  .debug_pubtypes 0 : { *(.debug_pubtypes) }
  .debug_ranges   0 : { *(.debug_ranges) }
  /* DWARF Extension.  */
  .debug_macro    0 : { *(.debug_macro) }
  .debug_addr     0 : { *(.debug_addr) }
  .ARM.attributes 0 : { KEEP (*(.ARM.attributes)) KEEP (*(.gnu.attributes)) }
  .note.gnu.arm.ident 0 : { KEEP (*(.note.gnu.arm.ident)) }
  /DISCARD/ : { *(.note.GNU-stack) *(.gnu_debuglink) *(.gnu.lto_*) }
}


==================================================
attempt to open /Volumes/Android/buildbot/src/android/binutils/prebuilts/ndk/current/platforms/android-21/arch-arm64/Volumes/Android/buildbot/src/android/binutils/out/binutils/darwin/arm64/install/binutils-arm64-darwin/aarch64-linux-android/lib64/libpthread.so failed
attempt to open /Volumes/Android/buildbot/src/android/binutils/prebuilts/ndk/current/platforms/android-21/arch-arm64/Volumes/Android/buildbot/src/android/binutils/out/binutils/darwin/arm64/install/binutils-arm64-darwin/aarch64-linux-android/lib64/libpthread.a failed
attempt to open /Volumes/Android/buildbot/src/android/binutils/prebuilts/ndk/current/platforms/android-21/arch-arm64/usr/local/lib64/libpthread.so failed
attempt to open /Volumes/Android/buildbot/src/android/binutils/prebuilts/ndk/current/platforms/android-21/arch-arm64/usr/local/lib64/libpthread.a failed
attempt to open /Volumes/Android/buildbot/src/android/binutils/prebuilts/ndk/current/platforms/android-21/arch-arm64/lib64/libpthread.so failed
attempt to open /Volumes/Android/buildbot/src/android/binutils/prebuilts/ndk/current/platforms/android-21/arch-arm64/lib64/libpthread.a failed
attempt to open /Volumes/Android/buildbot/src/android/binutils/prebuilts/ndk/current/platforms/android-21/arch-arm64/usr/lib64/libpthread.so failed
attempt to open /Volumes/Android/buildbot/src/android/binutils/prebuilts/ndk/current/platforms/android-21/arch-arm64/usr/lib64/libpthread.a failed
attempt to open /Volumes/Android/buildbot/src/android/binutils/prebuilts/ndk/current/platforms/android-21/arch-arm64/Volumes/Android/buildbot/src/android/binutils/out/binutils/darwin/arm64/install/binutils-arm64-darwin/aarch64-linux-android/lib/libpthread.so failed
attempt to open /Volumes/Android/buildbot/src/android/binutils/prebuilts/ndk/current/platforms/android-21/arch-arm64/Volumes/Android/buildbot/src/android/binutils/out/binutils/darwin/arm64/install/binutils-arm64-darwin/aarch64-linux-android/lib/libpthread.a failed
attempt to open /Volumes/Android/buildbot/src/android/binutils/prebuilts/ndk/current/platforms/android-21/arch-arm64/usr/local/lib/libpthread.so failed
attempt to open /Volumes/Android/buildbot/src/android/binutils/prebuilts/ndk/current/platforms/android-21/arch-arm64/usr/local/lib/libpthread.a failed
attempt to open /Volumes/Android/buildbot/src/android/binutils/prebuilts/ndk/current/platforms/android-21/arch-arm64/lib/libpthread.so failed
attempt to open /Volumes/Android/buildbot/src/android/binutils/prebuilts/ndk/current/platforms/android-21/arch-arm64/lib/libpthread.a failed
attempt to open /Volumes/Android/buildbot/src/android/binutils/prebuilts/ndk/current/platforms/android-21/arch-arm64/usr/lib/libpthread.so failed
attempt to open /Volumes/Android/buildbot/src/android/binutils/prebuilts/ndk/current/platforms/android-21/arch-arm64/usr/lib/libpthread.a failed
/Users/user/Library/Android/sdk/ndk-bundle/toolchains/aarch64-linux-android-4.9/prebuilt/darwin-x86_64/lib/gcc/aarch64-linux-android/4.9.x/../../../../aarch64-linux-android/bin/ld: cannot find -lpthread

Which NDK version do you use?

--- a/modules/mono/build_scripts/mono_configure.py
+++ b/modules/mono/build_scripts/mono_configure.py
@@ -163,7 +163,7 @@ def configure(env, env_mono):
 
             copy_file(mono_bin_path, '#bin', mono_dll_name + '.dll')
     else:
-        is_apple = (sys.platform == 'darwin' or "osxcross" in env)
+        is_apple = env['platform'] in ['osx', 'iphone']
 
         sharedlib_ext = '.dylib' if is_apple else '.so'
 

